commit c9f48cde0a22641ce3daf54596a9ecebdab91435
Author: Rich Felker <dalias@aerifal.cx>
Date:   Sat Oct 19 15:39:45 2019 -0400

    fix fpregset_t type on powerpc64
    
    the userspace ucontext API has this as an array rather than a
    structure.
    
    commit 3c59a868956636bc8adafb1b168d090897692532 fixed the
    corresponding mistake for vrregset_t, namely that the original
    powerpc64 port used a mix of types from 32-bit powerpc and powerpc64
    rather than matching the 64-bit types.

diff --git a/arch/powerpc64/bits/signal.h b/arch/powerpc64/bits/signal.h
index 2cc0604c..4dec22a5 100644
--- a/arch/powerpc64/bits/signal.h
+++ b/arch/powerpc64/bits/signal.h
@@ -9,11 +9,7 @@
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 
 typedef unsigned long greg_t, gregset_t[48];
-
-typedef struct {
-	double fpregs[32];
-	double fpscr;
-} fpregset_t;
+typedef double fpregset_t[33];
 
 typedef struct {
 #ifdef __GNUC__
